### ISSUE TITLE
'warning CS8655: The switch expression does not handle some null inputs' on suppressed nulls

### ISSUE THREAD

**Issue by alrz** [dotnet/roslyn#38586]
**Version Used**: master 

**Steps to Reproduce**:

```cs
#nullable enable
public class C {
    public int M0(C a)  => a switch { C _ => 0 }; // ok
    public int M1(C? a) => a switch { C _ => 0 }; // warns
    public int M2(C? a) => a! switch { C _ => 0 }; // warns (none expected)
}
```

**Expected Behavior**: no warnings in M2

**Actual Behavior**: compiler warns about unhandled null in M2
